app/src/components/pages/PlugAndPlay.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import Cookies from 'universal-cookie';
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Divider,
+  Grid,
+  MenuItem,
+  TextField,
+  Typography
+} from '@mui/material';
+import {
+  backendUrl,
+  backendUrlInverse,
+  company,
+  companyInverse
+} from 'components/shared/constants/plugandplay.constants';
+import LoadingSpinner from 'components/organisms/LoadingSpinner';
+
+const companies: company[] = [company.company3, company.company4];
+
+function PlugAndPlay() {
+  const cookies = new Cookies();
+  const [backend, setBackend] = useState('');
+  const [algorithm1, setAlgorithm1] = useState('');
+  const [algorithm2, setAlgorithm2] = useState('');
+  const [warningDialogOpen, setWarningDialogOpen] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [oldConfig, setOldConfig] = useState<any>({});
+
+  const getChanges = () => {
+    let changeStrings = [];
+    const oldBackend = backendUrl[oldConfig.REACT_APP_BACKEND_URL as keyof typeof backendUrl];
+    const oldAlgorithm1 = company[oldConfig.REACT_APP_ALGORITHM_1 as keyof typeof company];
+    const oldAlgorithm2 = company[oldConfig.REACT_APP_ALGORITHM_2 as keyof typeof company];
+
+    if (oldBackend !== backend) {
+      changeStrings.push(`Backend changed from ${oldBackend} to ${backend}\n`);
+    }
+    if (oldAlgorithm1 !== algorithm1) {
+      changeStrings.push(`Algorithm 1 changed from ${oldAlgorithm1} to ${algorithm1}\n`);
+    }
+    if (oldAlgorithm2 !== algorithm2) {
+      changeStrings.push(`Algorithm 2 changed from ${oldAlgorithm2} to ${algorithm2}\n`);
+    }
+    if (changeStrings.length === 0) {
+      changeStrings.push('No Changes');
+    }
+    return changeStrings;
+  };
+  const submitChanges = () => {
+    const config = {
+      headers: {
+        Accept: 'application/vnd.heroku+json; version=3',
+        Authorization: `Bearer ${process.env.REACT_APP_HEROKU}`,
+        'Content-Type': 'application/json'
+      }
+    };
+    const configVars = {
+      REACT_APP_BACKEND_URL: backendUrlInverse[backend as keyof typeof backendUrlInverse],
+      REACT_APP_ALGORITHM_1: companyInverse[algorithm1 as keyof typeof companyInverse],
+      REACT_APP_ALGORITHM_2: companyInverse[algorithm2 as keyof typeof companyInverse]
+    };
+    axios.patch('https://api.heroku.com/apps/seng499company4frontend/config-vars', configVars, config).then((res) => {
+      console.log(res);
+      cookies.remove('user');
+      window.location.href = '/';

```

Comment:
Nit: Could use 
const navigate = useNavigate();
navigate('/'); to align with the rest of the app's nav

Reply:
This was actually intentional, when I clear the cookie and then use navigate() it doesnt reload the app, so itll just navigate to the home page without re-checking the now cleared cookie. As a result you'll still be able to browse the app as if you did have the cookie set. Using the `.href` method forces a refresh too